<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title>Preprocessing Microbiome Sequences with QIIME 2 - A Guide to Importing, Trimming, Denoising, and Classifying 16S Suquencing Data | Ke Xu</title> <meta name="author" content="Ke Xu"> <meta name="description" content="A simple, whitespace theme for academics. Based on [*folio](https://github.com/bogoli/-folio) design. "> <meta name="keywords" content="jekyll, jekyll-theme, academic-website, portfolio-website"> <link href="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha256-DF7Zhf293AJxJNTmh5zhoYYIMs2oXitRfBjY+9L//AY=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@5.15.4/css/all.min.css" integrity="sha256-mUZM63G8m73Mcidfrv5E+Y61y7a12O5mW4ezU3bxqW4=" crossorigin="anonymous"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/academicons@1.9.1/css/academicons.min.css" integrity="sha256-i1+4qU2G2860dGGIOJscdC30s9beBXjFfzjWLjBRsBg=" crossorigin="anonymous"> <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/github.css" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="data:image/svg+xml,&lt;svg%20xmlns=%22http://www.w3.org/2000/svg%22%20viewBox=%220%200%20100%20100%22&gt;&lt;text%20y=%22.9em%22%20font-size=%2290%22&gt;%F0%9F%90%88&lt;/text&gt;&lt;/svg&gt;"> <link rel="stylesheet" href="/assets/css/main.css"> <link rel="canonical" href="https://xkcococo.github.io/blog/2023/Use-QIIME2-to-Preprocess-16S-Sequencing-Files/"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/jwarby/jekyll-pygments-themes@master/native.css" media="none" id="highlight_theme_dark"> <script src="/assets/js/theme.js"></script> <script src="/assets/js/dark_mode.js"></script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top"> <div class="container"> <a class="navbar-brand title font-weight-lighter" href="/"><span class="font-weight-bold">Ke </span>Xu</a> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item "> <a class="nav-link" href="/">About</a> </li> <li class="nav-item active"> <a class="nav-link" href="/blog/">Blog<span class="sr-only">(current)</span></a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">Projects</a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">Publications</a> </li> <li class="nav-item "> <a class="nav-link" href="/abstracts/">Abstracts</a> </li> <li class="nav-item "> <a class="nav-link" href="/teaching/">Teaching</a> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="fas fa-moon"></i> <i class="fas fa-sun"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5"> <div class="post"> <header class="post-header"> <h1 class="post-title">Preprocessing Microbiome Sequences with QIIME 2 - A Guide to Importing, Trimming, Denoising, and Classifying 16S Suquencing Data</h1> <p class="post-meta">January 12, 2023• Ke Xu</p> <p class="post-tags"> <a href="/blog/2023"> <i class="fas fa-calendar fa-sm"></i> 2023 </a>   ·   <a href="/blog/tag/qiime2"> <i class="fas fa-hashtag fa-sm"></i> QIIME2</a>   <a href="/blog/tag/16s"> <i class="fas fa-hashtag fa-sm"></i> 16S</a>   <a href="/blog/tag/sequencing"> <i class="fas fa-hashtag fa-sm"></i> sequencing</a>   <a href="/blog/tag/fastq"> <i class="fas fa-hashtag fa-sm"></i> fastq</a>   </p> </header> <article class="post-content"> <p><strong>The latest QIIME 2 tutorials version <a href="https://docs.qiime2.org/2022.11/" rel="external nofollow noopener" target="_blank">2022.11</a>.</strong> This tutorial is based on the latest version of QIIME 2, 2022.11. Before starting, it’s recommended to review the QIIME 2 tutorials to understand the different functions and plugins available.</p> <p><br> <strong>Step 1: Importing Data in QIIME2.</strong></p> <p>Before importing your data, it is important to identify the correct sequencing file format. The format of your sequencing files can be compared to the formats listed on <a href="https://docs.qiime2.org/2022.11/tutorials/importing/" rel="external nofollow noopener" target="_blank">this page</a>.</p> <p>For the purpose of this tutorial, let’s use <code class="language-plaintext highlighter-rouge">Casava 1.8 paired-end demultiplexed fastq</code> as an example. This format consists of two <code class="language-plaintext highlighter-rouge">fastq.gz</code> files for each sample, with names such as <code class="language-plaintext highlighter-rouge">L2S357_15_L001_R1_001.fastq.gz</code> and <code class="language-plaintext highlighter-rouge">L2S357_15_L001_R2_001.fastq.gz</code>. The <code class="language-plaintext highlighter-rouge">R1</code> and <code class="language-plaintext highlighter-rouge">R2</code> stand for one forward and one reverse sequencing file, respectively.</p> <p>The fields in the file name, separated by underscores, are: <code class="language-plaintext highlighter-rouge">sample identifier</code>, <code class="language-plaintext highlighter-rouge">barcode sequence or barcode identifier</code>, <code class="language-plaintext highlighter-rouge">lane number</code>, <code class="language-plaintext highlighter-rouge">read direction (i.e. R1 or R2)</code>, and <code class="language-plaintext highlighter-rouge">set number</code>.</p> <p>To import your data, run the following code:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>qiime tools import  \
  --type 'SampleData[PairedEndSequencesWithQuality]'  \
  --input-path /path/you/saved/sequencing/files/afog16s-all-modified/seq  \
  --input-format CasavaOneEightSingleLanePerSampleDirFmt  \
  --output-path /path/you/saved/your/output/demux-paired-end_demodata.qza
</code></pre></div></div> <p>This will import your demultiplexed paired-end sequences, ready for adapter trimming.</p> <p><br> <strong>Step 2: Searching Demultiplexed Paired-end Sequences For Adapters and Remove Them.</strong></p> <p>If adapters were used during PCR procedures, they need to be searched for and removed from all reads using Cutadapt. For the purposes of this tutorial, let’s suppose the adapter in all forward sequences is <code class="language-plaintext highlighter-rouge">GACTACCAGGGTATCTAATCCTGTTTGCTCCCC</code>, and the adapter in all reverse sequences is <code class="language-plaintext highlighter-rouge">CCTACGGGAGGCAGCAGTGGGGAATATTGCACAAT</code>.</p> <p>To trim the adapters, run the following code:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>qiime cutadapt trim-paired \
  --i-demultiplexed-sequences demux-paired-end_demodata.qza \
  --p-front-f GACTACCAGGGTATCTAATCCTGTTTGCTCCCC \
  --p-front-r CCTACGGGAGGCAGCAGTGGGGAATATTGCACAAT \
  --p-error-rate 0 \
  --o-trimmed-sequences /path/you/saved/your/output/trimmed-seq.qza \
  --verbose
</code></pre></div></div> <p>Let’s move on to the next step of denoising our trimmed sequence data.</p> <p><br> <strong>Step 3: Sequence Denoising with DADA2.</strong></p> <p>Before we denoise our sequence data, we need to visualize it using the <a href="https://view.qiime2.org/" rel="external nofollow noopener" target="_blank">QIIME 2 view</a>. This allows us to determine the parameters for denoising. To do this, we will use the following command:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>qiime demux summarize \
  --i-data trimmed-seq.qza \
  --o-visualization trimmed-seq.qzv
</code></pre></div></div> <p>We can then drop the trimmed-seq.qzv file into the view window. The parameters of <code class="language-plaintext highlighter-rouge">p-trim-left-f</code>, <code class="language-plaintext highlighter-rouge">p-trim-left-r</code>, <code class="language-plaintext highlighter-rouge">p-trunc-len-f</code>, and <code class="language-plaintext highlighter-rouge">p-trunc-len-r</code> will be determined based on the overall quality of the reads.</p> <p>To perform the denoising, we will use the following command:</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>qiime dada2 denoise-paired \
  --i-demultiplexed-seqs 'trimmed-seq.qza' \
  --p-trim-left-f 0 \
  --p-trim-left-r 0 \
  --p-trunc-len-f 220 \
  --p-trunc-len-r 200 \
  --o-representative-sequences rep-seqs-dada2.qza \
  --o-table table-dada2.qza \
  --o-denoising-stats satats-dada2.qza \
  --p-n-threads 24
</code></pre></div></div> <p>With this command, we have generated <code class="language-plaintext highlighter-rouge">table-dada2.qza</code> and <code class="language-plaintext highlighter-rouge">rep-seqs-dada2.qza</code> for creating the taxonomy table and feature table.</p> <p><br> <strong>Step 4: Generating the Taxonomy Table.</strong></p> <p>There are two options for generating the taxonomy table:</p> <ol> <li> <a href="https://docs.qiime2.org/2022.11/tutorials/feature-classifier/" rel="external nofollow noopener" target="_blank">Train your own feature classifier</a>, or 2. Use a <a href="https://docs.qiime2.org/2022.11/data-resources/" rel="external nofollow noopener" target="_blank">pre-traiend feature classifier</a> provided by QIIME 2.</li> </ol> <p>In this tutorial, we will use the pre-trained classifier “Silva 138 99% OTUs full-length sequences.”</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#download classifier
wget -O "silva-138-99-nb-classifier.qza" "https://data.qiime2.org/2020.8/common/silva-138-99-nb-classifier.qza"

qiime feature-classifier classify-sklearn \
  --i-classifier silva-138-99-nb-classifier.qza \
  --i-reads rep-seqs-dada2.qza \
  --o-classification taxonomy.qza

#Export taxonomy info in .tsv format
qiime tools export 
  --input-path taxonomy.qza \
  --output-path exported-feature-table
</code></pre></div></div> <p>With this, we have generated the taxonomy table in <code class="language-plaintext highlighter-rouge">.tsv</code> format.</p> <p><br> <strong>Step 5: Generating Feature Table for Each Sample.</strong></p> <p>In this step, we will use <code class="language-plaintext highlighter-rouge">table-dada2.qza</code> to generate our feature table.</p> <div class="language-plaintext highlighter-rouge"><div class="highlight"><pre class="highlight"><code>#Creating a TSV BIOM table
#first, export your data as a .biom
qiime tools export --input-path table-dada2.qza --output-path exported-feature-table


#Covert .biom to .tsv
biom convert -i exported-feature-table/feature-table.biom -o exported-feature-table/feature-table.tsv --to-tsv
biom head -i feature-table.tsv
</code></pre></div></div> <p><br> <strong>Step 6: Combining the Feature Table and Taxonomy Table.</strong></p> <p>At this point, you should have two output files: <code class="language-plaintext highlighter-rouge">taxonomy.tsv</code> and <code class="language-plaintext highlighter-rouge">feature-table.tsv</code>. These two tables can be combined by using the sequencing id as the key.</p> <p><br> <strong>You’re all set! You now have everything you need to perform your downstream statistical analysis. Enjoy!</strong></p> </article> </div> </div> <script async src="//busuanzi.ibruce.info/busuanzi/2.3/busuanzi.pure.mini.js"> </script> <footer class="fixed-bottom"> <div class="container mt-0"> © Copyright 2023 Ke Xu. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/bootstrap@4.6.1/dist/js/bootstrap.bundle.min.js" integrity="sha256-fgLAgv7fyCGopR/gBNq2iW3ZKIdqIcyshnUULC4vex8=" crossorigin="anonymous"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@4/imagesloaded.pkgd.min.js"></script> <script defer src="/assets/js/masonry.js" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.0.8/dist/medium-zoom.min.js" integrity="sha256-7PhEpEWEW0XXQ0k6kQrPKwuoIomz8R8IYyuU1Qew4P8=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js"></script> <script defer src="/assets/js/common.js"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script type="text/javascript">window.MathJax={tex:{tags:"ams"}};</script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.0/es5/tex-mml-chtml.js"></script> <script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script> <script async src="https://www.googletagmanager.com/gtag/js?id=G-WLXWD7REDQ"></script> <script>function gtag(){window.dataLayer.push(arguments)}window.dataLayer=window.dataLayer||[],gtag("js",new Date),gtag("config","G-WLXWD7REDQ");</script> <script type="text/javascript">function progressBarSetup(){"max"in document.createElement("progress")?(initializeProgressElement(),$(document).on("scroll",function(){progressBar.attr({value:getCurrentScrollPosition()})}),$(window).on("resize",initializeProgressElement)):(resizeProgressBar(),$(document).on("scroll",resizeProgressBar),$(window).on("resize",resizeProgressBar))}function getCurrentScrollPosition(){return $(window).scrollTop()}function initializeProgressElement(){let e=$("#navbar").outerHeight(!0);$("body").css({"padding-top":e}),$("progress-container").css({"padding-top":e}),progressBar.css({top:e}),progressBar.attr({max:getDistanceToScroll(),value:getCurrentScrollPosition()})}function getDistanceToScroll(){return $(document).height()-$(window).height()}function resizeProgressBar(){progressBar.css({width:getWidthPercentage()+"%"})}function getWidthPercentage(){return getCurrentScrollPosition()/getDistanceToScroll()*100}const progressBar=$("#progress");window.onload=function(){setTimeout(progressBarSetup,50)};</script> </body> </html>